lib/shopping_cart.rb

```diff
@@ -0,0 +1,43 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize (name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {name: name, capacity: capacity.delete("^0-9")}
+  end
+
+  def total_number_of_products
+    total = []
+    @products.each do |product|
+      total << product.quantity.to_i
+    end
+    total.sum
+  end
+
+  def is_full?
+    if total_number_of_products > @capacity.to_i
```

Comment:
you can just have this line without the `if` in this entire method since it will either evaluate to true or false automatically.